src/wallet/wallet.cpp

```diff
@@ -1921,14 +1862,30 @@ CAmount CWallet::GetDebit(const CTransactionBase& txBase, const isminefilter& fi
     return nDebit;
 }
 
-CAmount CWallet::GetCredit(const CTransactionBase& txBase, const isminefilter& filter) const
+CAmount CWallet::GetCredit(const CTransactionBase& txBase, const isminefilter& filter,
+                           bool& fCanBeCached, bool keepImmatureVoutsOnly = false) const
 {
+    // If al least one vout is immature, result cannot be cached
+    // Sum over mature vouts only or immature vouts only depending on keepImmatureVoutsOnly flag
+
     CAmount nCredit = 0;
-    for(const CTxOut& txout: txBase.GetVout()) {
-        nCredit += GetCredit(txout, filter);
+    fCanBeCached = true;
+    for(unsigned int pos = 0; pos < txBase.GetVout().size(); ++pos) {
+        if (txBase.IsCertificate()) { //optimization since non-certs have mature vouts only
+
+            if (!pcoinsTip->IsOutputMature(txBase.GetHash(), pos)) {
+                fCanBeCached = false;
+                if (!keepImmatureVoutsOnly) continue;
```

Comment:
This apparently is skipped only if flag is false. If flag is true an immature amount can be part of GetCredit computation below?

Reply:
Agree it was a bug. Fixed and extended sc_cert_base.py to prove it